src/anomalib/trainer/connectors/thresholding_connector.py

```diff
@@ -72,12 +71,13 @@ def compute(self):
             outputs (EPOCH_OUTPUT | List[EPOCH_OUTPUT]): Outputs are only used to check if the model has pixel level
                 predictions.
         """
-        if self.trainer.image_threshold is not None:
-            self.trainer.image_threshold.compute()
-        if self.trainer.task_type in (TaskType.SEGMENTATION, TaskType.DETECTION):
-            self.trainer.pixel_threshold.compute()
-        else:
-            self.trainer.pixel_threshold.value = self.trainer.image_threshold.value
+        if self.threshold_method == ThresholdMethod.ADAPTIVE:
+            if self.trainer.image_threshold is not None:
+                self.trainer.image_threshold.compute()
+            if self.trainer.task_type in (TaskType.SEGMENTATION, TaskType.DETECTION):
+                self.trainer.pixel_threshold.compute()
+            else:
+                self.trainer.pixel_threshold.value = self.trainer.image_threshold.value
 
```

Comment:
This part will change with the thresholding refactor, right?

Reply:
Yes